cmd/manager/main_test.go

```diff
@@ -23,3 +34,50 @@ func TestCheckIfRequiredFilesExist(t *testing.T) {
 		"Expected error message is absent")
 
 }
+
+// TestIsValidKubernetesVersion tests if isValidKubernetes function throws error if K8s version is less than v1.18.0
+func TestIsValidKubernetesVersion(t *testing.T) {
+	fakeConfigClient := fakeconfigclient.NewSimpleClientset()
+	// specify version to be tested
+	fakeConfigClient.Discovery().(*fakediscovery.FakeDiscovery).FakedServerVersion = &version.Info{
+		GitVersion: "v1.17.0",
+	}
+	err := isValidKubernetesVersion(fakeConfigClient)
+	require.Error(t, err, "Function isValidKubernetes did not throw an error when it was expected to")
+	assert.Contains(t, err.Error(), "invalid server version: v1.17.0.Supported versions >= v1.18.0")
+}
+
+// TestNetworkConfiguration tests if validate() function throws error for invalid network type and throws error if
+// hybrid overlay is not enabled.
+func TestNetworkConfiguration(t *testing.T) {
```

Comment:
Is it possible to make this [table driven](https://github.com/golang/go/wiki/TableDrivenTests)?